### PR TITLE
Port virtual_staining examples into applications/cytoland

### PR DESCRIPTION
**Pull request by edyoshikun**
## Summary

- Ports the `examples/virtual_staining/` tree from `main` into `applications/cytoland/examples/` on top of `modular-viscy-staging`, covering four tutorial groups: VS_model_inference, dlmbl_exercise, vcp_tutorials, phase_contrast.
- Rewrites every `viscy.*` import to the modular package layout (`viscy_data`, `viscy_transforms`, `viscy_utils`, `cytoland.engine`, `viscy_utils.callbacks.HCSPredictionWriter`, `viscy_utils.losses.MixedLoss`, `viscy_utils.evaluation.metrics.mean_average_precision`).
- Updates `setup.sh` in dlmbl_exercise and phase_contrast to `pip install -e "applications/cytoland[metrics]"` instead of the retired top-level `viscy[metrics,visual,examples]>=0.2` wheel.
- Adds `**/examples/**` to `[tool.ruff] per-file-ignores` for `D`, `E402`, `E501`, `F821` so jupytext percent-cell scripts lint cleanly without forcing docstrings, top-of-file imports, short lines, or importing notebook builtins.
- Adds a Tutorials and demos section to `applications/cytoland/README.md` pointing at the new tree.

Notebooks (`.ipynb`) are intentionally not ported — regenerate from `solution.py` / `*.py` with `jupytext --to ipynb` when needed.

## Import migration

| Old | New |
|---|---|
| `viscy.data.hcs` | `viscy_data.hcs` |
| `viscy.transforms` | `viscy_transforms` |
| `viscy.trainer` | `viscy_utils.trainer` |
| `viscy.translation.engine.{VSUNet, FcmaeUNet, AugmentedPredictionVSUNet}` | `cytoland.engine.{...}` |
| `viscy.translation.engine.MixedLoss` | `viscy_utils.losses.MixedLoss` |
| `viscy.translation.predict_writer.HCSPredictionWriter` | `viscy_utils.callbacks.HCSPredictionWriter` |
| `viscy.translation.evaluation_metrics.mean_average_precision` | `viscy_utils.evaluation.metrics.mean_average_precision` |

## Test plan

- [x] `python -c "import ast; ast.parse(open(f).read())"` passes for all ported `.py` files
- [x] `ruff check` + `ruff format` pass via pre-commit on every commit
- [x] Import smoke test — `uv run --package cytoland python -c "from cytoland.engine import ...; from viscy_data.hcs import ...; ..."` resolves every ported symbol
- [ ] End-to-end run of any demo against real data + checkpoints (requires download from `public.czbiohub.org/comp.micro/viscy` and a GPU; not part of this PR)

### CODE REVIEW

File: applications/cytoland/README.md [hunk 1]
```diff
@@ -37,6 +37,21 @@ data:
...
+| Folder | What it demonstrates |
+|--------|----------------------|
+| [`examples/VS_model_inference/`](./examples/VS_model_inference/) | Python API inference demos for VSCyto2D, VSCyto3D, VSNeuromast, and TTA-augmented sliding-window prediction |
+| [`examples/vcp_tutorials/`](./examples/vcp_tutorials/) | Virtual Cell Platform quick-start and organism-specific walkthroughs (HEK293T, neuromast) |
+| [`examples/dl-course-exercise/`](./examples/dl-course-exercise/) | Image-translation course exercise (training from scratch + evaluation) — used at DL@MBL and DL@Janelia |
+| [`examples/configs/`](./examples/configs/) | YAML configs for `viscy fit` / `viscy predict` across models (VSCyto2D/3D, VSNeuromast, FNet3D, dynacell) |
+
+All demo scripts are written as jupytext-style percent-cell `.py` files.
+Regenerate paired `.ipynb` notebooks with `jupytext --to ipynb solution.py`
+if you prefer the notebook UI.
```
[the remark this answers is not in the document]
Comment (edyoshikun) [reply applications/cytoland/README.md:54]: this is actually not needed.

File: applications/cytoland/examples/phase_contrast/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+# Overview:
+
+Generalization to Zernike phase contrast images. This demo showcases the use of VSCyto3D model with and without augmentations on Zernike phase contrast data.
+
+## Setup
+
+Run the setup script to create the environment for this exercise and download the dataset.
+```bash
+source setup.sh
```
[the remark this answers is not in the document]
Comment (edyoshikun) [reply applications/cytoland/examples/phase_contrast/README.md]: this should be relative to just the examples for now.

File: applications/cytoland/examples/dl-course-exercise/README.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+into `$DATA_ROOT`. Typical runtime is 20–40 min. It does **not** create a
+Python environment — students do that themselves with `setup_student.sh`.
+
+## Use VSCode
+
+Install VSCode and the Python + Jupyter extensions, then open
+[`solution.py`](solution.py) and pick the **Python (06_image_translation)**
+kernel from the top-right kernel selector. The script uses
+[cell mode](https://code.visualstudio.com/docs/python/jupyter-support-py), so
+you can execute each `# %%` block interactively.
```
[the remark this answers is not in the document]
Comment (edyoshikun) [reply applications/cytoland/examples/dl-course-exercise/README.md]: perhaps we remove this or do we need more explicit instructions.

File: applications/cytoland/examples/phase_contrast/setup.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+# Install ipykernel in the environment.
+conda install -y ipykernel nbformat nbconvert ruff jupytext ipywidgets --name vs_Phc
+# Specifying the environment explicitly.
+# conda activate sometimes doesn't work from within shell scripts.
+
+# Install cytoland (pulls in viscy-data, viscy-models, viscy-transforms, viscy-utils).
+# Run this from the root of the VisCy monorepo checkout.
+# Find path to the environment - conda activate doesn't work from within shell scripts.
+ENV_PATH=$(conda info --envs | grep vs_Phc | awk '{print $NF}')
+$ENV_PATH/bin/pip install -e "applications/cytoland[metrics]"
```
[the remark this answers is not in the document]
Comment (edyoshikun) [reply applications/cytoland/examples/phase_contrast/setup.sh]: oksuggest modification

File: applications/cytoland/examples/VS_model_inference/demo_vscyto_w_ttas.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    .to(DEVICE)
+    .eval()
+)
+
+# Load data
+path = Path("/path/to/your.zarr/0/1/000000")
+with open_ome_zarr(path) as ds:
+    vol_np = np.asarray(ds.data[0:1, 0:1])  # (1, 1, Z, Y, X)
+
+vol = torch.from_numpy(vol_np).float().to(DEVICE)
```
[the remark this answers is not in the document]
Comment (edyoshikun) [reply applications/cytoland/examples/VS_model_inference/demo_vscyto_w_ttas.py:66]: update with the precomputed fov stats uring NormalizedSampled

File: applications/cytoland/examples/phase_contrast/solution.py [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+)
+model_VSCyto3D_no_augmentation.eval()
+# Model with augmentation
+model_VSCyto3D_w_augmentation = VSUNet.load_from_checkpoint(
+    VSCyto3D_model_ckpt, architecture="UNeXt2", model_config=config_VSCyto3D
+)
+model_VSCyto3D_w_augmentation.eval()
+
+# Setup the Trainer
+trainer = VisCyTrainer(accelerator="gpu", devices=[GPU_ID], precision="16-mixed")
```
[the remark this answers is not in the document]
Comment (edyoshikun) [reply applications/cytoland/examples/phase_contrast/solution.py]: make sure the model and inference is run in the GPU

File: applications/cytoland/examples/dl-course-exercise/solution.py [hunk 7]
```diff
@@ -0,0 +1,2980 @@
...
+
+# Run inference
+with torch.inference_mode():
+    predicted_phase = fluor2phase_model(fluor_input)[0]
+
+fluor_input = fluor_input.cpu().numpy()
+predicted_image = predicted_phase.cpu().numpy().squeeze(0)
+target_phase = rescale_intensity(target_image[0, 0], out_range=(0, 1))
+predicted_phase = rescale_intensity(predicted_image[0, 0], out_range=(0, 1))
+ssim_phase = metrics.structural_similarity(target_phase, predicted_phase, data_range=1)
```
[the remark this answers is not in the document]
Comment (edyoshikun) [reply applications/cytoland/examples/dl-course-exercise/solution.py:2083]: move the import to here if this isthe first declartion

File: applications/cytoland/examples/dl-course-exercise/solution.py [hunk 8]
```diff
@@ -0,0 +1,2980 @@
...
+
+# Normalize data range to 0-1
+target_nuc[0] = rescale_intensity(target_nuc[0], in_range="image", out_range=(0, 1))
+single_pred_nuc[0] = rescale_intensity(single_pred_nuc[0], in_range="image", out_range=(0, 1))
+target_mem[0] = rescale_intensity(target_mem[0], in_range="image", out_range=(0, 1))
+single_pred_mem[0] = rescale_intensity(single_pred_mem[0], in_range="image", out_range=(0, 1))
+target_nuc[0] = rescale_intensity(target_nuc[0], in_range="image", out_range=(0, 1))
+tta_pred_nuc[0] = rescale_intensity(tta_pred_nuc[0], in_range="image", out_range=(0, 1))
+tta_pred_mem[0] = rescale_intensity(tta_pred_mem[0], in_range="image", out_range=(0, 1))
+tta_pred_nuc[0] = rescale_intensity(tta_pred_nuc[0], in_range="image", out_range=(0, 1))
```
[the remark this answers is not in the document]
Comment (edyoshikun) [reply applications/cytoland/examples/dl-course-exercise/solution.py]: move the TTA implementation before the compute the predicionts in the same cell
